push-messaging-and-notifications/main.js

```diff
@@ -1,23 +1,64 @@
 'use strict';
 
-var API_KEY = '<YOUR API KEY>';
+var API_KEY = window.GoogleSamples.Config.gcmAPIKey;
+var GCM_ENDPOINT = 'https://android.googleapis.com/gcm/send';
 
 var curlCommandDiv = document.querySelector('.js-curl-command');
 var isPushEnabled = false;
 
+// This method handles the removal of subscriptionId
+// in Chrome 44 by concatenating the subscription Id
+// to the subscription endpoint
+function endpointWorkaround(pushSubscription) {
+  // Make sure we only mess with GCM
+  if (pushSubscription.endpoint.indexOf('https://android.googleapis.com/gcm/send') !== 0) {
+    return pushSubscription.endpoint;
+  }
+
+  var mergedEndpoint = pushSubscription.endpoint;
+  // Chrome 42 + 43 will not have the subscriptionId attached
+  // to the endpoint.
+  if (pushSubscription.subscriptionId &&
+    !pushSubscription.endpoint.indexOf(pushSubscription.subscriptionId) !== -1) {
```

Comment:
Can you switch the double-negation to `pushSubscription.endpoint.indexOf(pushSubscription.subscriptionId) === -1`?